functions/index.ts

```diff
@@ -245,3 +246,26 @@ functions.database.ref('gamePortal/gameSpec/reviews/{reviewedGameSpecId}/{review
       return starsSummary;
     });
 });
+
+exports.resizeImage =
+functions.storage.object().onChange.onChange(async (event:any) => {
+  console.log("Resizing the image");
+  // Extract file data
+  const path = event.data.name;
+  const bucket = event.data.bucket;
+  const [imgType, uid, id] = path.split("/");
+
+  if(imgType !== "upload") return;
+
+  // Download file into tmp directory
+  const tmpFilePath = `/tmp/${id}.jpg`;
+  const file = bucket.file(path);
+  console.log("Downloading image to: ", tmpFilePath);
+  await file.download({ destination : tmpFilePath});
+
+  // Resize it and upload it to cloud storage
+  console.log("Resizing image and uploading to cloud storage...");
```

Comment:
Let's generate 3 images:
1) An image of exactly the same size and similar type (type is either png/jpg), but using quality=70 (see about quality: http://www.imagemagick.org/script/command-line-options.php?#quality )
2) Same as 1, with quality=50
2) A thumbnail image of lower size (at most 50x50), and always with type=JPG

We can't always use jpg, because for JPG: "It also does NOT allow any form of transparency." (see http://www.imagemagick.org/Usage/thumbnails/)
And if the input is a png, we need to preserve transparency. 

Add comments to the function saying what's the input and what's the output.
E.g., given that a user uploaded an image to:
gs://universalgamemaker.appspot.com/images/-KtgKdhyzZ6HrXsGwcpi.png
we create images in:
gs://universalgamemaker.appspot.com/quality70/-KtgKdhyzZ6HrXsGwcpi.png
gs://universalgamemaker.appspot.com/quality50/-KtgKdhyzZ6HrXsGwcpi.png
gs://universalgamemaker.appspot.com/thumbnail/-KtgKdhyzZ6HrXsGwcpi.jpg